CEP 37: `conda-lock.yml` lockfiles

jaimergp opened the pull request:
## Checklist for submitter

- [X] I am submitting a new CEP: [`conda-lock.yml` lockfiles](https://github.com/jaimergp/ceps/blob/conda-lock/cep-XXXX.md).
  - [X] I am using the CEP template by creating a copy `cep-0000.md` named `cep-XXXX.md` in the root level.
- [ ] I am submitting modifications to CEP XX. <!-- reflect CEP number here -->
- [ ] Something else: (add your description here).

<!-- delete section below if this is not a new CEP -->
## Checklist for CEP approvals

- [ ] The vote period has ended and the vote has passed the necessary quorum and approval thresholds.
- [ ] A new CEP number has been minted. Usually, this is `${greatest-number-in-main} + 1`.
- [ ] The `cep-XXXX.md` file has been renamed accordingly.
- [ ] The `# CEP XXXX - ` header has been edited accordingly.
- [ ] The CEP status in the table has been changed to approved.
- [ ] The last modification date in the table has been updated accordingly.
- [ ] The table in the README has been updated with the new CEP entry.
- [ ] The `pre-commit` checks are passing.


<!-- Just as a reminder, everyone in all conda org spaces (including PRs)
     must follow the Conda Org Code of Conduct (link below).

     Finally, once again, thanks for your time and effort. If you have any
     feedback in regards to your experience contributing here, please
     let us know!

     Helpful links:
       - Conda Org COC: https://github.com/conda/governance/blob/main/CODE_OF_CONDUCT.md
-->

jaimergp commented:
cc @maresb, lmk what you think!

maresb commented on cep-XXXX.md:
```diff
@@ -0,0 +1,197 @@
+
+Required.
+
+A dictionary that describes provenance and reproducibility metadata for the lockfile. It MUST only include the following keys, mapping to the values specified below. Additional keys MUST NOT be present.
```
I'm not sure if this is technically true. It's morally true though.

maresb reviewed (commented):
I've taken an initial quick pass and this looks pretty good. Ping @basnijholt.

I think it may be helpful to provide the following additional context:

There's a "unique" constraint on the `package` list with key `(name, manager, platform, category)`.

Currently known producers:
* `conda-lock lock`, https://github.com/basnijholt/pixi-to-conda-lock

Currently known consumers:
* `conda-lock install`, `micromamba env create`

jaimergp commented:
Thanks for the prompt review!


> There's a "unique" constraint on the `package` list with key `(name, manager, platform, category)`.

Added in [ee2dff0](https://github.com/conda/ceps/pull/138/commits/ee2dff0121f57579a2c52b3e2124102497420588).

> Actually md5 and sha256 are technically optional.

Reworded in [a6d58bc](https://github.com/conda/ceps/pull/138/commits/a6d58bc37e53abb4dbf95389a3ad3de571b4bdc8).

> Currently known producers:

Thanks! Added in [1202623](https://github.com/conda/ceps/pull/138/commits/1202623cd6ff17deb7d2a04f3cdee3a5d339647f). Note that [`conda-lockfiles`](https://github.com/conda-incubator/conda-lockfiles) is also a thing now.

maresb commented on cep-XXXX.md:
```diff
@@ -0,0 +1,197 @@
+When these two strings are concatenated, it MUST result in a:
+
+- For `manager == "conda"` entries, a `MatchSpec` string, as described in [CEP PR#82](https://github.com/conda/ceps/pull/82).
+- For `manager == "pip"` entries, a [PEP 440](https://peps.python.org/pep-0440/) dependency specifier.
```
Thanks so much for checking! That's not intentional, but I think it's also not used for anything beyond walking the dependency graph for assigning categories, so I think it won't lead to any actual issues.

jaimergp commented:
Dear @conda/steering-council,

The vote for this CEP has started. It will be open for two weeks, until **March 2nd, 2026, 23:59 Anywhere on Earth**. This time period has been chosen to make it eligible for time-out rules. As an Enhancement Proposal vote, it requires 60% affirmative votes to pass.

To vote, please mark the relevant checkbox under your username:

- Uwe Korn (@xhochy)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Christopher J. 'CJ' Wright (@cj-wright)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Marius van Niekerk (@mariusvniekerk)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Cheng H. Lee (@chenghlee)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Filipe Fernandes (@ocefpaf)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Marcelo Duarte Trevisani (@marcelotrevisani)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Michael Sarahan (@msarahan)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Marcel Bargull (@mbargull)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- John Kirkham (@jakirkham)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Jannis Leidel (@jezdez)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Wolf Vollprecht (@wolfv)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Jaime Rodríguez-Guerra (@jaimergp)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Bas Zalmstra (@baszalmstra)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Hind Montassif (@Hind-M)
  - [ ] yes
  - [ ] no
  - [ ] abstain
- Pavel Zwerschke (@pavelzw)
  - [ ] yes
  - [ ] no
  - [ ] abstain

jaimergp commented:
The voting period has concluded. This is a CEP vote, so it requires 60% of the steering council to vote to reach quorum, and a 60% of affirmative votes to pass. However, since it was open for more than two weeks and three reminders were sent (actually, four), it also qualifies for timeout rules, where the quorum threshold is simply five people.

- Size of the steering council: 15
  - Number of votes: 13 -> 86.67%
    - Yes: 12 -> 100%
    - No: 0
    - Abstain: 1
  - Did not vote: 2

This CEP has passed 🎉.
